doc/api/dns.md

```diff
@@ -568,8 +568,10 @@ dns.setServers([
 
 An error will be thrown if an invalid address is provided.
 
-The `dns.setServers()` method must not be called while a DNS query is in
-progress.
+The `dns.setServers()` method affects only the methods of [`dns.Resolver`][]
+that have equivalents as `dns.*()` APIs (and specifically *not*
+[`dns.lookup()`][]). It must not be called while those methods are in progress.
```

Comment:
I'd note that `dns.lookup()` uses [getaddrinfo](http://man7.org/linux/man-pages/man3/getaddrinfo.3.html) which has no concept of servers or TTL.

Reply:
> I'd note that dns.lookup() uses getaddrinfo 

That seems out of place here in the setServers() API doc. If you follow the link to dns.lookup(), you will find it links to implementation considerations, and it is noted there that lookup is implemented with getaddrinfo. See https://nodejs.org/api/dns.html#dns_dns_lookup

> has no concept of servers or TTL

getaddrinfo uses the servers configured in resolv.conf, the docs for which are linked from the implementation notes.

TTL is returned by DNS servers, not sure what you mean by that, perhaps you mean timeout on lookup? that is configurable with resolv.conf

Reply:
> it is noted there that lookup is implemented with getaddrinfo

Oh, sorry, I didn't see that.

> TTL is returned by DNS servers, not sure what you mean by that, perhaps you mean timeout on lookup?

These are not my words and even that I understand them correctly.

> dns.lookup() uses getaddrinfo which has no concept of servers<sup>1</sup> or TTL<sup>2</sup>.

<sup>1</sup> `dns.setServers(...)` doesn't affect `dns.lookup(...)` as it's OS layer.
<sup>2</sup> `dns.lookup(...)` doesn't have any TTL as it's OS layer.

Reply:
Anyway, let's leave that as it is :D